private_gpt/server/chat/chat_service.py

```diff
@@ -97,8 +98,11 @@ def _chat_engine(
         system_prompt: str | None = None,
         use_context: bool = False,
         context_filter: ContextFilter | None = None,
+        context_template: str | None = None,
     ) -> BaseChatEngine:
         if use_context:
+            if context_template is None:
+                context_template = settings().rag.default_context_template
```

Comment:
You should avoid calling settings() directly, and instead inject settings as a dependency in the ChatService constructor. Check how settings are injected in EmbeddingComponent or LLMComponent for example.